### PR TITLE
#162948910 Users should be able to reset password via email

### PR DESCRIPTION
**Pull request by boltonotieno**
#### What does this PR do?

Allows the user to reset their password 

#### Description of Task to be completed?

* User should request to reset their password
* A  link should be sent to their email address
* From the link, they should be able to reset their password
#### How should this be manually tested?

* Clone ah-alpha repo $ git clone https://github.com/andela/ah-alpha.git
* git checkout `ft-user-reset-password-162948910`
* Run `pip install -r requirements.txt` to update requirements.txt 
* set up postgres database on the psql command line
* Create a virtual environment and create a .env file and add the database credentials
* export database credentials. please note that there are no spaces before and after =
``` export DB_NAME=" name of your database"
      export DB_PASSWORD="passord to your database user"
      export DB_HOST="your local server eg (127.0.0.1)"
      export DB_USER="your database username eg (postgres)"
```
* Note- if you have no password set for your database leave the password field blank example: 
  `export DB_PASSWORD=""`

* In the project's root directory run `python manage.py runserver`
* Create a user by sending a POST request to the url http://127.0.0.1:8000/api/users/ with the following in the body.
```
{
    "user":{
        "username":"Username",
        "email":"example@email.com",
        "password":"Password"
    }
}
```
* Request for your password reset by sending a POST request to  the link http://127.0.0.1:8000/api/users/password_request/ with the following in the body.
```
{
    "user":{
        "email":"your email"
    }
}
```
- You will receive a link to reset your password in your email address
- copy the link and paste on postman
- send a PATCH request to the copied link 
- In the body tag use the format below to change your password
```
{
    "user":{
        "password":new password"
    }
}
```
#### What are the relevant pivotal tracker stories?
[#162948910 ](https://www.pivotaltracker.com/story/show/162948910 )

### CODE REVIEW

File: authors/apps/authentication/backends.py [hunk 1]
```diff
@@ -1,10 +1,36 @@
...
+
+        payload = jwt_payload_handler(user)
+        token = jwt_encode_handler(payload)
+
+        return token
+
+"""Configure JWT Here"""
+
+
+class GetAuthentication(TokenAuthentication):
```
Comment (tesh254) [inline authors/apps/authentication/backends.py:32]: Introduce doc strings for every method, function and class

File: authors/apps/authentication/models.py [hunk 2]
```diff
@@ -98,16 +101,16 @@ def __str__(self):
...
-      return self.username
+        """
+        This method is required by Django for things like handling emails.
+        Typically, this would be the user's first and last name. Since we do
+        not store the user's real name, we return their username instead.
+        """
+        return self.username
 
     def get_short_name(self):
         """
```
Comment (tesh254) [inline authors/apps/authentication/models.py]: Reduce this space here

File: authors/apps/authentication/backends.py (hunk 1 above)
Comment (Ruiru11) [reply authors/apps/authentication/backends.py:32]: On it

File: authors/apps/authentication/models.py (hunk 2 above)
Comment (Ruiru11) [reply authors/apps/authentication/models.py]: Working on this

**Review by jamesbeamie — APPROVED** [review]

File: authors/apps/authentication/models.py [hunk 3]
```diff
@@ -4,117 +4,121 @@
 
 from django.conf import settings
 from django.contrib.auth.models import (
-		AbstractBaseUser, BaseUserManager, PermissionsMixin
+        AbstractBaseUser, BaseUserManager, PermissionsMixin
 )
-from django.db import models 
+from django.db import models
+# from rest_framework_jwt.settings import api_settings
```
Comment (frankip) [inline authors/apps/authentication/models.py]: @BoltC0rt3z remove unused code

File: authors/apps/authentication/tests/test_password_request.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        """Test request password reset without an email"""
+        response = self.submit_email(self.user_email[0])
+
+        self.assertEqual(
+            response.status_code,
+            status.HTTP_400_BAD_REQUEST)
+
+        self.assertEqual(
+            response.json(),
+            {"message": 'Please provide an email'})
```
Comment (frankip) [inline authors/apps/authentication/tests/test_password_request.py]: @BoltC0rt3z Nice work on testing the error messages but you can import the messages file to assert the response messages in your tests

File: authors/apps/authentication/models.py (hunk 3 above)
Comment (boltonotieno) [reply authors/apps/authentication/models.py]: Its done

File: authors/apps/authentication/tests/test_password_request.py (hunk 4 above)
Comment (boltonotieno) [reply authors/apps/authentication/tests/test_password_request.py]: Thank you for the feedback. I have worked on it.

**Review by danuluma — APPROVED** [review]

**Pull request merged**